src/main/java/com/mongodb/kafka/connect/sink/WriteConcernException.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.kafka.connect.sink;
+
+import java.util.Formatter;
+import java.util.Locale;
+
+import com.mongodb.bulk.WriteConcernError;
+
+/**
+ * The {@linkplain #getMessage() message} {@linkplain Formatter format} is {@code "v=1, code=%d,
+ * codeName=%s, message=%s, details=%s"}, where {@code details} is JSON produced with {@link
+ * org.bson.BsonDocument#toJson()}. We may change it in the future, in which case the version
+ * (marked with {@code v}) will be incremented.
+ */
+final class WriteConcernException extends NoStackTraceDlqException {
+  private static final long serialVersionUID = 1L;
+
+  WriteConcernException(final WriteConcernError error) {
+    super(
+        String.format(
+            Locale.ENGLISH,
+            "v=1, code=%d, codeName=%s, message=%s, details=%s",
```

Comment:
Suggestion set v1 to be static of the class.

Reply:
I thought about it before submitting the PR, and it appears that specifying it as part of the format string literal is better because otherwise the format will look like `"v=%d, ..."`, and it is likely that the docs team would specify it that way in the official docs, and we would have to clarify in the doc ticket that the format should be documented as `"v=1, ..."`. It's a minor negative thing, but since I did not find any positive things regarding using the `"v=%d, ..."` format in the code, I decided to have the value of the version as part of the format literal.

What benefits do you see with regard to using the `"v=%d, ..."` format in the code, and having the value version as a `static` field?

Reply:
Only benefit was to make it a forcing function to ensure that a version is set, also a variable is more self documenting that it being hidden in the string. The end result to a user of course would be the same. Its more to help with understanding the code in 6 months or for a new developer to understand what it means.

Reply:
Done. Extracted it to a static field.